tests/cephadm/test_cephadm_upgrade.py

```diff
@@ -42,6 +43,30 @@ def run(ceph_cluster, **kwargs) -> int:
     config = kwargs["config"]
     config["overrides"] = kwargs.get("test_data", {}).get("custom-config")
     verify_image = bool(config.get("verify_cluster_health", False))
+    if config.get("args", {}).get("rhcs-version", None):
+        _rhcs_version = config.get("args").get("rhcs-version", None)
+        _rhcs_release = config.get("args").get("release", None)
+        rhbuild = config.get("rhbuild")
+        _platform = "-".join(rhbuild.split("-")[1:])
+        _base_url, _registry, _image_name, _image_tag = fetch_build_artifacts(
+            _rhcs_release, _rhcs_version, _platform
+        )
```

Comment:
This works well for `cephfs` however this may not be the case for NVMe-oF or NFS-Ganesha or SMB wherein we need to look for tags of non-base image.

I believe we would need to expand `fetch_build_artifacts` or write a new one say `fetch_image_manifest` that holds the complete list of images tagged for that particular release.

What we have observed is that the base version can be from one release (7.1 - GA - 7.1.0) whereas the rest of the images would be from `7.1z3`. Though this would be the case for customers, may not have the right set of environment.

If the `latest_container_info` does hold these details then we should add them.